tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/ComputeSystemsListViewModel.cs

```diff
@@ -101,21 +99,22 @@ public override string ToString()
         return AccessibilityName;
     }
 
-    public ComputeSystemsListViewModel(ComputeSystemsLoadedData loadedData)
+    public ComputeSystemsListViewModel(
+        ComputeSystemProviderDetails providerDetails,
+        KeyValuePair<DeveloperIdWrapper, ComputeSystemsResult> devIdComputeSystemPair)
     {
-        Provider = loadedData.ProviderDetails.ComputeSystemProvider;
-        DevIdToComputeSystemMap = loadedData.DevIdToComputeSystemMap;
-
-        // Get the first developerId and compute system result.
-        var devIdToResultKeyValuePair = DevIdToComputeSystemMap.FirstOrDefault();
-        CurrentResult = devIdToResultKeyValuePair.Value;
-        CurrentDeveloperId = devIdToResultKeyValuePair.Key;
+        Provider = providerDetails.ComputeSystemProvider;
+        CurrentResult = devIdComputeSystemPair.Value;
+        CurrentDeveloperId = devIdComputeSystemPair.Key;
 
         DisplayName = Provider.DisplayName;
 
         if ((CurrentResult != null) && (CurrentResult.ComputeSystems != null))
         {
-            ComputeSystems = CurrentResult.ComputeSystems.Select(computeSystem => new ComputeSystemCache(computeSystem)).ToList();
+            // Filter out compute systems that do not support configuration.
```

Comment:
Double checking: Won't filtering out affect the Environments page as well?
Also, since it is being filtered out, consider renaming the class or adding a comment.

Reply:
This file is located in `tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/ComputeSystemsListViewModel.cs` in the setup flow. Its separate from the environments page. Only the ones that support configuration are ever used in it so thats why I moved the check here. Both the environments and the setupflow retrieve separate lists of compute systems and they can manipulate the list how ever they want. I think i'll keep it as is for now since it's within the setupflow name space.